pkg/input/golden_test/tf/regula-issue-389.json

```diff
@@ -0,0 +1,34 @@
+{
+  "format": "",
+  "format_version": "",
+  "input_type": "tf_hcl",
+  "environment_provider": "iac",
+  "meta": {
+    "filepath": "golden_test/tf/regula-issue-389/main.tf"
+  },
+  "resources": {
+    "aws_s3_bucket": {
+      "aws_s3_bucket.test": {
+        "id": "aws_s3_bucket.test",
+        "resource_type": "aws_s3_bucket",
+        "namespace": "golden_test/tf/regula-issue-389/main.tf",
+        "meta": {},
+        "attributes": {
+          "bucket_prefix": "testytest",
+          "tags": [
+            "owner:tf",
+            "service:var.service",
```

Comment:
Is it possible to only do this if the variable is a string type? Otherwise I think we could produce some confusing errors and behavior. For example with this configuration:

```hcl
variable "foo" {
  type = bool

}

resource "aws_s3_bucket" "name" {
  bucket = var.foo ? "foo" : "bar"
}
```

If you don't provide a value for `foo`, this is the error you get with the current version:

```
null-var.tf: evaluate: error: null-var.tf:7,12-19: Null condition; The condition value is null. Conditions must either be true or false.
```

vs after these changes:

```
null-var.tf: evaluate: error: null-var.tf:7,12-19: Incorrect condition type; The condition expression must be of type bool.
```

Reply:
It turns out to be quite hard to determine the type for these, as this code path is also used for other things that can be "missing".  Instead, I added just a bit more logging so it's at least clear what happens:

```
> go build && ./policy-engine fixture -v test.tf
6:19PM WRN test.tf: Missing term for "var.foo", substituted in as string
6:19PM WRN test.tf: evaluate: error: test.tf:7,12-19: Incorrect condition type; The condition expression must be of type bool
```